### PR TITLE
adding hazel/chaz

### PR DESCRIPTION
**Pull request by jasonhazel**
## Description

Adding initial configuration and default keymap for [Chaz](https://github.com/hazels-garage/chaz)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hazel/chaz/info.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    "bootloader": "caterina",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "nkro": false
+    },
+    "matrix_size": {
+        "cols": 5,
+        "rows": 8
+    },
```
Comment (zvecr) [inline keyboards/hazel/chaz/info.json:15]:
```suggestion
```

File: keyboards/hazel/chaz/info.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    "matrix_pins": {
+        "cols": ["B5", "B4", "E6", "D7", "C6"],
+        "rows": ["D3", "D1", "D4", "F5", "D2", "D0", "F6", "F4"]
+    },
+    "processor": "atmega32u4",
```
Comment (zvecr) [inline keyboards/hazel/chaz/info.json:20]:
```suggestion
```

File: keyboards/hazel/chaz/info.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
+{
+    "manufacturer": "Hazel",
+    "keyboard_name": "Chaz",
+    "maintainer": "jasonhazel",
+    "url": "https://github.com/hazels-garage/chaz",
+    "bootloader": "caterina",
```
Comment (zvecr) [inline keyboards/hazel/chaz/info.json:6]:
```suggestion
```

File: keyboards/hazel/chaz/rules.mk [hunk 4]
```diff
@@ -0,0 +1 @@
+CAPS_WORD_ENABLE = yes
```
Comment (zvecr) [inline keyboards/hazel/chaz/rules.mk:1]: We generally dont want this enabled by default (nor mixing rules.mk/info.json enabling of features)
```suggestion
# This file intentionally left blank
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/hazel/chaz/rules.mk (hunk 4 above)
Comment (jasonhazel) [reply keyboards/hazel/chaz/rules.mk]: changes commited, and I also removed `DOUBLE_TAP_SHIFT_TURNS_ON_CAPS_WORD` from default `config.h`

**Pull request closed without merging**